src/tag.ts

```diff
@@ -4,11 +4,11 @@ export default class Tag {
   constructor(public tag: string) {}
 
   async getSubject(): Promise<string> {
-    return await this.getTagContents('subject')
+    return await this.getTagContents('contents:subject')
   }
 
-  async getBody(): Promise<string> {
-    return await this.getTagContents('body')
+  async getBody(includeSubject: boolean): Promise<string> {
+    return await this.getTagContents(includeSubject ? 'contents' : 'contents:body')
   }
 
   private async getTagContents(contents: string): Promise<string> {
```

Comment:
Great idea. I'll implement that once you decide on the correct terminology (`name` vs. `title`).

Reply:
I just remembered something that I forgot in my initial implementation of those changes.

When you just use `%(contents)` and your tag is GPG-signed, it includes that `-----BEGIN PGP SIGNATURE-----` block at the bottom as well.

And as far as I know there is no filter for just that built into git. I solved it like this for myself:

```bash
git tag -v --format='%(contents)' TAG_NAME | awk '/^-*BEGIN PGP SIGNATURE-*$/ {exit} {print}'
```

This gives me the entire thing minus the signature.

This of course also works, if that block is missing, so adding that to the actual `git` call should work in any case.

Reply:
Good catch! I'll spend some time researching different ways to filter out the signature.

Reply:
OK, so a couple of ideas.

## Bad Ideas

- `--format=%(contents:lines=1000)` will include the subject and body, but exclude the signature
- `--format=%(contents:signature)` will get the signature, and you can use `String.replace`

## Workable Idea

```shell
git tag -v --format='%(contents:subject)

%(contents:body)' $TAG_NAME
```
This will basically rebuild the tag message. Sadly, the code won't look as clean and pretty :laughing:

******

Here's what I'm thinking:
```typescript
// Let's just require a filter for now...
getContents(filter: string): string;
getContents(filter: string[]): string[];
getContents(filter: string | string[]): string | string[] {
  if (typeof filter === 'string') {
    // str = ...
    return str;
  }
  return filter.map((s: string) => this.getContents(s));
}
```

See [function overloads](https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads), and let me know if you have any questions.

Then, when `tag-as-title` is `true`, you call `getContents(['subject', 'body'])`, and join with newlines.

Reply:
I like it, but I am having trouble implementing the overloaded function. I tried out the codespace you suggested, but the linter keeps telling me about `getContents` being a duplicate name. :confused:  I must be doing something wrong. If you don't mind making these adjustments yourself and pushing them into this branch, that would probably be more efficient. I am on board with all your suggested changes.

Reply:
That works for me. I'll get to it when I have time.
I've never *formally* learned Typescript, and [sometimes the typing system baffles me](https://github.com/jamiebuilds/json-parser-in-typescript-very-bad-idea-please-dont-use), so I may have given incorrect instructions :shrug:

Reply:
@daniil-berg You weren't doing anything wrong, it was a [false catch](https://github.com/typescript-eslint/typescript-eslint/issues/291).

https://github.com/spenserblack/actions-tag-to-release/pull/114/commits/5412c9b2a9c38d26650d4cfc9b0d106ac2d78391